lib/ImportSource.php

```diff
@@ -121,12 +124,15 @@ public function copyToFolder(Folder $destination, string $sourcePath): bool {
 						$file = $destination->get($path);
 						if ($file instanceof File) {
 							$file->putContent($stream);
+							$file->touch($stat['mtime']);
 						} else {
 							$file->delete();
-							$destination->newFile($path, $stream);
+							$file = $destination->newFile($path, $stream);
+							$file->touch($stat['mtime']);
 						}
 					} catch (NotFoundException $e) {
-						$destination->newFile($path, $stream);
+						$file = $destination->newFile($path, $stream);
+						$file->touch($stat['mtime']);
```

Comment:
Put the `$file->touch` call after the try catch rather than having it in 3 places